src/duqtools/ids/_mapping.py

```diff
@@ -22,69 +24,82 @@ def insert_re_caret_dollar(string: str) -> str:
     return string
 
 
-def recursive_defaultdict():
-    """Recursive defaultdict."""
-    return defaultdict(recursive_defaultdict)
-
-
-def defaultdict_to_dict(ddict: defaultdict):
-    """defaultdict_to_dict, turns a nested defaultdict into a nested dict.
-
-    Parameters
-    ----------
-    ddict : defaultdict
-        ddict
-    """
-
-    # Convert members to dict first
-    for k, v in ddict.items():
-        if isinstance(v, defaultdict):
-            ddict[k] = defaultdict_to_dict(v)
-    # Finally convert self to dict
-    return dict(ddict)
-
-
 class IDSMapping(Mapping):
 
     def __init__(self, ids, exclude_empty: bool = True):
         self._ids = ids
         self.exclude_empty = exclude_empty
 
-        # All fields in the core profile in a single dict
-        self.flat_fields: dict = {}
-
-        # All fields, in the core profile in a nested dict
-        self.fields: dict = defaultdict(recursive_defaultdict)
+        # All available data fields are stored in this set.
+        self._keys: Set[str] = set()
 
         self.dive(ids, [])
-        self.fields = defaultdict_to_dict(self.fields)
 
     def __repr__(self):
         s = f'{self.__class__.__name__}(\n'
-        for key in self.fields.keys():
+        for key in self._keys:
             s += f'  {key} = ...\n'
         s += ')\n'
 
         return s
 
     def __getitem__(self, key: str):
-        try:
-            return self.flat_fields[key]
-        except KeyError:
-            pass
-        try:
-            return self.fields[key]
-        except KeyError:
-            raise
+        if key not in self._keys:
+            raise KeyError(key)
+
+        *parts, attr = key.split('/')
+
+        pointer = self._ids
+
+        for part in parts:
+            try:
+                i = int(part)
+            except ValueError:
```

Comment:
Shouldn't this information be available in the current pointer, something like this:
```python
  isinstance(pointer,list)
```

Reply:
You would think so, unfortunately the IMAS array does not follow convention. It does not implement some of the methods for it to be *Sequence-like*, for example: https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes